src/aleo_schnorr/mod.rs

```diff
@@ -0,0 +1,138 @@
+        ALEO_SCHNORRR_TAG,
+        KEYGEN_TAG,
+        secret_recovery_key,
+        session_nonce,
+    )?;
+
+    let private_key = PrivateKey::new(&mut rng).map_err(|_| TofnFatal)?;
+    let aleo_account = Account::try_from(private_key).map_err(|_| TofnFatal)?;
+
+    Ok(KeyPair { aleo_account })
+}
+
+pub fn sign<N: Network>(
+    signing_key: &KeyPair<N>,
+    msg_to_sign: &MessageDigest,
+) -> TofnResult<BytesVec> {
+    let message = aleo_encoded(msg_to_sign).map_err(|_| TofnFatal)?;
+
+    let sign = signing_key
+        .aleo_account
+        .sign(&message, &mut rand_chacha::ChaChaRng::from_entropy())
+        .map_err(|_| TofnFatal)?;
+
+    Ok(sign.to_string().as_bytes().to_vec())
+}
+
+pub fn verify<N: Network>(
+    address: &str,
+    signature: &Signature<N>,
+    message: &MessageDigest,
+) -> TofnResult<bool> {
+    use snarkvm::prelude::Address;
+
+    let message = aleo_encoded(message).map_err(|_| TofnFatal)?;
+
+    let address = Address::from_str(address).map_err(|_| TofnFatal)?;
+
+    Ok(signature.verify(&address, &message))
+}
+
+fn aleo_encoded<N: Network>(data: &MessageDigest) -> TofnResult<Vec<Field<N>>> {
+    let data = data.as_ref().try_into().map_err(|_| TofnFatal)?;
+    let num = cosmwasm_std::Uint256::from_le_bytes(data);
+    let message = format!("{num}group");
+
+    snarkvm::prelude::Value::from_str(message.as_str())
+        .map_err(|_| TofnFatal)?
+        .to_fields()
+        .map_err(|_| TofnFatal)
+}
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr as _;
+
+    use super::*;
+    use crate::sdk::key::dummy_secret_recovery_key;
+
+    pub type CurrentNetwork = snarkvm::prelude::TestnetV0;
+
+    #[test]
+    fn keygen_sign_decode_verify() {
+        let message = [
+            30, 165, 51, 99, 240, 22, 44, 209, 224, 46, 25, 4, 49, 49, 114, 238, 209, 48, 186, 136,
+            95, 224, 128, 254, 19, 109, 54, 40, 214, 206, 187, 13,
+        ]
+        .into();
+
+        let key_pair = keygen::<CurrentNetwork>(&dummy_secret_recovery_key(42), b"tofn nonce")
+            .map_err(|_| TofnFatal)
+            .unwrap();
+        let encoded_signature = sign(&key_pair, &message).map_err(|_| TofnFatal).unwrap();
+        let signature = Signature::<CurrentNetwork>::from_str(
+            &String::from_utf8(encoded_signature)
+                .map_err(|_| TofnFatal)
+                .unwrap(),
+        )
+        .map_err(|_| TofnFatal)
+        .unwrap();
+        let success = verify(
+            key_pair.encoded_verifying_key().as_str(),
+            &signature,
+            &message,
+        )
+        .map_err(|_| TofnFatal)
+        .unwrap();
+
+        assert!(success);
+    }
+
+    /// Check keygen/signing outputs against golden files to catch regressions (such as on updating deps).
+    /// Golden files were generated from tofn commit corresponding to tofnd v0.10.1 release
+    #[test]
+    fn keygen_sign_known_vectors() {
+        // todo!()
+    }
```

Comment:
Do we plan on implementing this?

If not, I'd suggest we remove this test item entirely. 

If we are planning to implement it, I suggest using the [ignored attribute](https://doc.rust-lang.org/reference/attributes/testing.html#the-ignore-attribute). This approach is more transparent since the test's ignored status will be visible during test runs, which is better than having a silent no-op test.

```suggestion
    /// Check keygen/signing outputs against golden files to catch regressions (such as on updating deps).
    /// Golden files were generated from tofn commit corresponding to tofnd v0.10.1 release
    #[test]
    #[ignore = "not yet implemented"]
    fn keygen_sign_known_vectors() { }
```